packages/blocks/src/api/registration.js

```diff
@@ -373,3 +373,22 @@ export const registerBlockStyle = ( blockName, styleVariation ) => {
 		};
 	} );
 };
+
+/**
+ * Unregisters a block style variation for the given block.
+ *
+ * @param {string} blockName          Name of block (example: “core/latest-posts”).
+ * @param {string} styleVariationName Name of class applied to the block.
+ */
+export const unregisterBlockStyle = ( blockName, styleVariationName ) => {
+	addFilter( 'blocks.registerBlockType', `${ blockName }/${ styleVariationName }/unregister`, ( settings, name ) => {
+		if ( blockName !== name ) {
+			return settings;
+		}
+
+		return {
+			...settings,
+			styles: remove( get( settings, [ 'styles' ], [] ), ( style ) => styleVariationName !== style.name ),
```

Comment:
`remove` is mutative, which we should avoid. Consider `reject` or `filter` instead:

- https://lodash.com/docs/4.17.10#remove
- https://lodash.com/docs/4.17.10#reject

Reply:
Reading this, it seems like it should be the opposite equality check, removing those variations where the style name matches the function argument.